### PR TITLE
[Feature] translate

### PR DESCRIPTION
**Pull request by YinZheng-Sun**
## Problem Summary:
Fixes # [22703](https://github.com/StarRocks/starrocks/issues/22704)

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
    - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by imay — COMMENTED** [review]
You can add more tests in sql-test that is in the test directory.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by YinZheng-Sun** [comment]
OK , but I wonder if translate should support utf-8 characters?

File: test/sql/test_function/R/test_translate [hunk 1]
```diff
@@ -0,0 +1,27 @@
+-- name: test translate
+select translate('abcdefg', 'abcdefg', '');
```
Comment (ZiheLiu) [inline test/sql/test_function/R/test_translate:2]: Add some test cases:
- which contains non-ASCII characters.
- or some parameters are invalid, such as non-constant or non-varchar `from_string` and `to_string`

File: be/src/exprs/string_functions.cpp [hunk 2]
```diff
@@ -2139,6 +2139,101 @@ StatusOr<ColumnPtr> StringFunctions::strcmp(FunctionContext* context, const Colu
...
+    }
+
+    return Status::OK();
+}
+
+StatusOr<ColumnPtr> StringFunctions::translate(FunctionContext* context, const Columns& columns) {
+    const ColumnPtr& arg0 = columns[0];
+    if (arg0->only_null()) {
+        return arg0;
+    }
```
Comment (ZiheLiu) [inline be/src/exprs/string_functions.cpp:2197]: if  the arg1 or arg2 is null, the result should also be null.

**Comment by ZiheLiu** [comment]
> OK , but I wonder if translate should support utf-8 characters?

@YinZheng-Sun Yes, since the format of `VARCHAR` in SR is UTF-8. 
For now in this PR, `translate` only considers the ASCII characters, but it cannot deal with the non-ASCII characters, each of which consists of multiple bytes.
For example, `SELECT TRANSLATE('测a试', '测试', '12')` should return `1a2`.

IMO, we should implement two versions of `TRANSLATE`: one is for `from_string` and `to_string` which only contains ASCII characters, and the other is for those contains non-ASCII characters.

You could see `StringFunctions::pad_prepare` for inspiration:

```C++
Status StringFunctions::pad_prepare(FunctionContext* context, FunctionContext::FunctionStateScope scope) {

    state->fill = ColumnHelper::get_const_value<TYPE_VARCHAR>(pad_column);
    // `fill_utf8_index` stores the boundary index of each character in `fill`.
    // `fill_is_utf8==false` indicates that `fill` only contains ASCII characters.
    state->fill_is_utf8 = state->fill.size > get_utf8_index(state->fill, &state->fill_utf8_index);
}
```
and `pad_const_not_null`:
```C++
    if (src_is_utf8 && pad_state->fill_is_utf8) {
        return pad_utf8_const<true, true, pad_type>(columns, src, (uint8_t*)fill.data, fill.size, len,
                                                    pad_state->fill_utf8_index);
    } else if (src_is_utf8) {
        return pad_utf8_const<true, false, pad_type>(columns, src, (uint8_t*)fill.data, fill.size, len,
                                                     pad_state->fill_utf8_index);
    } else if (pad_state->fill_is_utf8) {
        return pad_utf8_const<false, true, pad_type>(columns, src, (uint8_t*)fill.data, fill.size, len,
                                                     pad_state->fill_utf8_index);
    } else {
        return ascii_pad_ascii_const<pad_type>(columns, src, (uint8_t*)fill.data, fill.size, len);
    }
```

**Comment by YinZheng-Sun** [comment]
THX，I will have a try

**Comment by wangsimo0** [comment]
hi @YinZheng-Sun hows everything going :)

**Pull request closed without merging**